.travis/build-deploy.sh

```diff
@@ -60,9 +60,12 @@ elif travis_isPullRequest; then
     log_info "This is a pull-request build"
     ./mvnw verify $MVN_BUILD_FLAGS
 	(
-	    set +e
-	    log_info "Running danger"
-	    bundle exec danger --verbose
+            set +e
+            # if ${TRAVIS_BRANCH} is master, the script will fail since there is local master branch
+            git checkout -b ${TRAVIS_BRANCH} origin/${TRAVIS_BRANCH}
```

Comment:
If the branch already exists, this would fail, e.g.

```
$ git checkout -b master origin/master
fatal: A branch named 'master' already exists.
```

So, we need to check for the existence of the local branch first.

I'd also prefer, not to checkout the branch (thus changing the local files) and then checkout back to FETCH_HEAD again... this changes the files unnecessary.

We should use here simply `git branch` to create a branch. However, it has the same problem, if the branch already exists:

```
$ git branch master origin/master 
fatal: A branch named 'master' already exists.
```

This https://stackoverflow.com/questions/5167957/is-there-a-better-way-to-find-out-if-a-local-git-branch-exists might help here.

Reply:
I did a bit more tests now, and it seems, we need to fetch the remote branch origin/pmd/7.0.x first, before we can create a local branch (regardless whether checkout or branch is used):

```
$ git checkout -b pmd/7.0.x origin/pmd/7.0.x
fatal: 'origin/pmd/7.0.x' is not a commit and a branch 'pmd/7.0.x' cannot be created from it

$ git branch pmd/7.0.x origin/pmd/7.0.x
fatal: Not a valid object name: 'origin/pmd/7.0.x'.
```

So, I'd propose these commands:
```
if ! git show-ref --verify --quiet refs/heads/${TRAVIS_BRANCH}; then
    git fetch --no-tags origin +refs/heads/${TRAVIS_BRANCH}:refs/remotes/origin/${TRAVIS_BRANCH}
    git branch ${TRAVIS_BRANCH} origin/${TRAVIS_BRANCH}
fi
```

It's no problem, if the fetch command is executed multiple times for the same branch. So it could also be outside of the if-statement.

Reply:
Thank for your proposal! Done.